src/storage/app-state/slices/snackbarSlice.ts

```diff
@@ -0,0 +1,130 @@
+/*
+Copyright 2018 - 2022 The Alephium Authors
+This file is part of the alephium project.
+
+The library is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+The library is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with the library. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+import { createSlice, PayloadAction } from '@reduxjs/toolkit'
+
+import { SnackbarMessage } from '@/components/SnackbarManager'
+import i18n from '@/i18n'
+import { syncAddressesData } from '@/storage/app-state/slices/addressesSlice'
+import { contactStoredInPersistentStorage } from '@/storage/app-state/slices/contactsSlice'
+import {
+  apiClientInitFailed,
+  apiClientInitSucceeded,
+  customNetworkSettingsSaved
+} from '@/storage/app-state/slices/networkSlice'
+
+const sliceName = 'snackbarSlice'
+
+interface SnackbarSliceState {
+  messages: SnackbarMessage[]
+  offlineMessageWasVisibleOnce: boolean
+}
+
+const initialState: SnackbarSliceState = {
+  messages: [],
+  offlineMessageWasVisibleOnce: false
+}
+
+const displayError = (state: SnackbarSliceState, action: PayloadAction<string>) =>
+  displayMessageImmediately(state, { text: action.payload, type: 'alert', duration: 5000 })
+
+const snackbarSlice = createSlice({
+  name: sliceName,
+  initialState,
+  reducers: {
+    snackbarDisplayTimeExpired: (state) => {
+      if (state.messages.length > 0) state.messages.shift()
+    },
+    passwordValidationFailed: (state) =>
+      displayMessageImmediately(state, { text: i18n.t('Invalid password'), type: 'alert' }),
+    copiedToClipboard: (state) => displayMessageImmediately(state, { text: i18n.t('Copied to clipboard!') }),
+    transactionBuildFailed: displayError,
```

Comment:
Discussion : I'm really not sure about listing all the uses for snackbar messages in one place, especially since it's still needed to provide the message text.

As a developper who hasn't written the initial code, it's not intuitive to me to find the needed action here: `import { transactionBuildFailed } from '@/storage/app-state/slices/snackbarSlice'`.

I'd rather go for a more generic action like `snackbarMessageShown`.

As always, I'm open to be convinced otherwise! :)